test/e2e/tracing_test.go

```diff
@@ -104,32 +91,30 @@ var _ = Describe("Tracing", func() {
 			}, timeout, interval).Should(Succeed())
 		})
 
-		It("Should be able to send traces to the OTLP endpoint", func() {
-			shutdown, err := initProvider("localhost:4317")
-			Expect(err).ShouldNot(HaveOccurred())
-			defer shutdown(context.Background())
-			tracer := otel.Tracer("otlp-load-tester")
-
-			rootCtx, rootSpan := tracer.Start(ctx, "root", trace.WithAttributes(commonAttributes...))
-			spanIDs = append(spanIDs, rootSpan.SpanContext().SpanID().String())
-			traceID = rootSpan.SpanContext().TraceID().String()
-			rootSpan.End()
+		It("Should verify end-to-end trace delivery", func() {
+			traceID := newTraceID()
+			var spanIDs []pcommon.SpanID
 			for i := 0; i < 100; i++ {
-				_, span := tracer.Start(rootCtx, "non-root", trace.WithAttributes(commonAttributes...))
-				spanIDs = append(spanIDs, span.SpanContext().SpanID().String())
-				span.End()
+				spanIDs = append(spanIDs, newSpanID())
 			}
-		})
+			attrs := pcommon.NewMap()
+			attrs.PutStr("attrA", "chocolate")
+			attrs.PutStr("attrB", "raspberry")
+			attrs.PutStr("attrC", "vanilla")
+
+			traces := makeTraces(traceID, spanIDs, attrs)
 
-		It("Should verify traces were sent to the backend", func() {
+			sendTraces(context.Background(), traces, "localhost", 4317)
+
+			attrs.PutBool("bool", true)
```

Comment:
Is attribute modification _after_ sending the traces intentional?

Reply:
Nope, that change was reverted